linux/utils/check_guest_os_gui.yml

```diff
@@ -1,16 +1,63 @@
 # Copyright 2021-2022 VMware, Inc.
 # SPDX-License-Identifier: BSD-2-Clause
 ---
+# Check whether guest OS has desktop environment and
+# get display manager if deskotp environment is installed"
+- name: "Set fact of guest OS having desktop environment and display manager"
+  ansible.builtin.set_fact:
+    guest_os_with_gui: false
+    guest_os_display_manager: ""
+
 # Set default guest_os_with_gui by checking Xorg exists or not
 - name: "Check Xorg exists or not"
   ansible.builtin.shell: "type Xorg"
   ignore_errors: true
   delegate_to: "{{ vm_guest_ip }}"
   register: xorg_result
 
-- name: "Set fact of guest OS with/without desktop experience"
+- name: "Set fact of guest OS having desktop environment"
+  ansible.builtin.set_fact:
+    guest_os_with_gui: true
+  when:
+    - xorg_result is defined
+    - xorg_result.rc is defined
+    - xorg_result.rc  == 0
+
+- name: "Get guest OS display manager for {{ guest_os_ansible_distribution }}"
+  block:
+    - name: "Get guest OS display manager service detail"
+      ansible.builtin.shell: "systemctl status display-manager.service"
+      register: "display_manager_status"
+      changed_when: false
+      ignore_errors: true
+      delegate_to: "{{ vm_guest_ip }}"
+
+    - name: "Set fact of guest OS display manager"
+      ansible.builtin.set_fact:
+        guest_os_display_manager: |-
+          {%- if "GNOME Display Manager" in display_manager_status.stdout_lines[0] -%}gdm
+          {%- elif "Light Display Manager" in display_manager_status.stdout_lines[0] -%}lightdm
+          {%- elif "X Display Manager" in display_manager_status.stdout_lines[0] -%}xdm
+          {%- elif "LXDE Display Manager" in display_manager_status.stdout_lines[0] -%}lxdm
+          {%- elif "Simple Desktop Display Manager" in display_manager_status.stdout_lines[0] -%}sddm
+          {%- endif -%}
+      when:
+        - display_manager_status is defined
+        - display_manager_status.stdout_lines is defined
+        - display_manager_status.stdout_lines | length > 0
+  when:
+    - guest_os_with_gui | bool
+    - guest_os_ansible_distribution != 'Astra Linux (Orel)'
+
+
```

Comment:
What if `guest_os_display_manager` is empty?

Reply:
guest_os_display_manager is only used when enabling user auto login. If it is empty, user auto login will not be enabled, and ovt_verify_status would fail. I just added several fail messages in enable_auto_login.yml to identify the failure earlier when guest_os_display_manager is empty or not found display manager config file.